### PR TITLE
[kinetic] add emulation_clock publisher - cpp variant

### PR DESCRIPTION
**Pull request by fmessmer**
seems a bit more performant than #232

### CODE REVIEW

**Comment by fmessmer** [comment]
:warning: https://answers.ros.org/question/346686/disable_rostime-for-roscpp/

**Comment by fmessmer** [comment]
when implementing the emulation_clock timer with a boost timer instead of a ros timer, the cpp variant works....cpu load is approx. half of python variant...

however, I don't like that I have to specify  dt in milliseconds (i.e. specify as int) instead of dt in seconds (i.e. specify as double)...but I could not find a better way to implement the boost timer....
@benmaidel @LoyVanBeek @floweisshardt 
any better idea that would allow us to specify dt in seconds, i.e. as double?

still, the performance limit i.e. being able to emulate mbl scenario is at emulation_time_factor of <5.0 - mainly due to navigation not able to process its callbacks in time....

File: cob_hardware_emulation/scripts/emulation_clock.py [hunk 1]
```diff
@@ -7,16 +7,16 @@
...
         self.t = time.time()
-        self.dt = 0.01
+        self.dt_ms = rospy.get_param('/emulation_dt_ms', 10)
         self.time_factor = rospy.get_param('/emulation_time_factor', 1.0)
         if not self.time_factor > 0.0:
             rospy.logerr("emulation_time_factor must be >0.0, but is {}. exiting...".format(self.time_factor))
             exit(-1)
         self.pub = rospy.Publisher('/clock', Clock, queue_size=1)
-        rospy.Timer(rospy.Duration(self.dt), self.timer_cb)
+        rospy.Timer(rospy.Duration(self.dt_ms/1000.0), self.timer_cb)
```
Comment (LoyVanBeek) [inline cob_hardware_emulation/scripts/emulation_clock.py:16]: The official docs say the 'secs' arg or `rospy.Duration` should be an int, so no need for the `.0` at the end of `1000.0` if we're strict. Of course, everyone uses it a float which work in Python.

File: cob_hardware_emulation/src/emulation_clock.cpp [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+          timer_(io, boost::posix_time::milliseconds(dt_ms))
+    {
+        t_ = (double)std::time(NULL);
+        pub_ = nh_.advertise<rosgraph_msgs::Clock>("/clock", 1);
+        timer_.async_wait(boost::bind(&EmulationClock::timer_cb, this));
+    }
+
+    void timer_cb()
+    {
+        t_ += time_factor_*dt_ms_/1000;
```
Comment (LoyVanBeek) [inline cob_hardware_emulation/src/emulation_clock.cpp:25]: Some `()` would be good here maybe. 
Not that the calculation changes, but I'm not sure what happens with the double/int conversions going on under the hood.

**Comment by LoyVanBeek** [comment]
Specifying dt in millseconds is something I can live with, as long as it's absolutely clear that I need to specify milliseconds. 
That is the case via the parameter name.

**Pull request merged**